ansible_collections/arista/avd/plugins/plugin_utils/utils/avd_logging.py

```
@@ -0,0 +1,305 @@
+            task_vars = args[0] if args else kwargs.get("task_vars", {})
+            loggers_to_target = target_loggers or ["ansible_collections.arista.avd"]
+
+            # Prepare handlers, filters, and format based on knobs and task arguments
+            temp_handlers = []
+            if self._task.args.get("save_logs", False):
+                temp_handlers.append(SaveToResultHandler(result_dict=self.result))
+
+            # Build the context data and log format string dynamically
+            context_data, format_parts = {}, []
+            if add_role_context:
+                context_data["role_name"] = task_vars.get("ansible_role_name")
+                format_parts.append("[%(role_name)s] -")
+            if add_hostname_context:
+                context_data["hostname"] = task_vars.get("inventory_hostname")
+                format_parts.append("<%(hostname)s>")
+
+            format_parts.append("%(message)s")
+            log_format = " ".join(format_parts)
+
+            temp_filters = [ContextFilter(context_data)] if context_data else []
+
+            remove_global_handler = not self._task.args.get("live_display", True)
+
+            # Use the context manager to apply changes and ensure cleanup
+            with (
+                warnings.catch_warnings(record=True) as captured_warnings,
+                avd_logging_context(
+                    logger_names=loggers_to_target,
+                    temp_handlers=temp_handlers,
+                    temp_filters=temp_filters,
+                    log_format=log_format,
+                    remove_global_handler=remove_global_handler,
+                ),
+            ):
+                try:
+                    final_result = func(self, *args, **kwargs)
+                except BaseException as exc:
+                    # Recast errors as AnsibleActionFail
+                    msg = f"Error during plugin execution: {exc}"
+                    raise AnsibleActionFail(msg) from exc
```

That means we have to remove the try-except blocks we have in our plugins and let them crash so it's handled here.